packages/block-library/src/cover/edit/inspector-controls.js

```diff
@@ -284,6 +322,13 @@ export default function CoverInspectorControls( {
 								/>
 							</ToolsPanelItem>
 						) }
+						{ ! useFeaturedImage && !! imageSizeOptions.length && (
+							<ResolutionTool
+								value={ sizeSlug }
+								onChange={ updateImage }
+								options={ imageSizeOptions }
+							/>
```

Comment:
The reset behaviour for this control seems pretty odd. 

I think it might be missing the `defaultValue` prop otherwise it shows as having been modified before the user touches it. It's also used when resetting the control individually. 

If the default value prop isn't provided it will fall back to the first option in the available sizes which in my testing was thumbnail. That is different than how this panel's `resetAll` is resetting the `sizeSlug`.

Reply:
It should be working fine now, i have set default value to "Full Size"

[Edit-Post-“Test”-‹-gutenberg-—-WordPress.webm](https://github.com/user-attachments/assets/00110e65-fd8c-46e8-b70f-f775c18a5183)